libraries/botframework-connector/tests/auth/managedIdentityAuthenticator.test.js

```diff
@@ -0,0 +1,117 @@
+        });
+
+        testInput.forEach(({ name, value }) => {
+            it(`should throw if appId is ${name}`, function () {
+                assert.throws(() => new ManagedIdentityAuthenticator(value, testAudience, tokenProvider), {
+                    name: 'AssertionError',
+                    message: errorMsgAppId,
+                });
+            });
+        });
+
+        testInput.forEach(({ name, value }) => {
+            it(`should throw if audience is ${name}`, function () {
+                assert.throws(() => new ManagedIdentityAuthenticator(testAppId, value, tokenProvider), {
+                    name: 'AssertionError',
+                    message: errorMsgAudience,
+                });
+            });
+        });
+
+        it('should throw with null tokenProviderFactory', function () {
+            assert.throws(() => new ManagedIdentityAuthenticator(testAppId, testAudience, null), {
+                name: 'AssertionError',
+                message: errorMsgToken,
+            });
+        });
+    });
+
+    describe('getToken', function () {
+        it('should get token', async function () {
+            const sut = new ManagedIdentityAuthenticator(testAppId, testAudience, tokenProvider);
+            const token = await sut.getToken();
+
+            assert.strictEqual(token.token, authResult.token);
+            assert.strictEqual(token.expiresOnTimestamp, authResult.expiresOnTimestamp);
+        });
+
+        it('should retry and acquire token', async function () {
+            let callsToAcquireToken = 0;
+            const tokenProvider = new JwtTokenProviderFactory(testAppId);
+            tokenProvider.createAzureServiceTokenProvider = sinon.spy(() => ({
+                getToken() {
+                    callsToAcquireToken++;
+                    if (callsToAcquireToken === 1) {
+                        throw new Error();
+                    }
+
+                    return authResult;
+                },
+            }));
+            const sut = new ManagedIdentityAuthenticator(testAppId, testAudience, tokenProvider);
+            const token = await sut.getToken();
+
+            assert.strictEqual(token.token, authResult.token);
+            assert.strictEqual(token.expiresOnTimestamp, authResult.expiresOnTimestamp);
+            assert.strictEqual(callsToAcquireToken, 2);
+        });
+
+        it('should retry on error', async function () {
+            const maxRetries = 5;
+            let callsToAcquireToken = 0;
+            const tokenProvider = new JwtTokenProviderFactory(testAppId);
+            tokenProvider.createAzureServiceTokenProvider = sinon.spy(() => ({
+                getToken() {
+                    callsToAcquireToken++;
+                    throw new Error('AuthError');
+                },
+            }));
+
+            const sut = new ManagedIdentityAuthenticator(testAppId, testAudience, tokenProvider);
+
+            try {
+                await sut.getToken();
```

Comment:
I think these tests are better written with `await assert.rejects()`, like so:

```typescript
await assert.rejects(sut.getToken(), {
  message: 'AuthError'
});
assert.strictEqual(...);
```

This code won't actually fail if no error is thrown, which it seems like it should.